### PR TITLE
[IMP]added boolean field, to avoid iterate over many2many field

### PR DESCRIPTION
**Pull request by hugho-ad**
Dummy pr https://github.com/Vauxoo/yoytec/pull/396
IMPS:
- [x]  recustion function removed from template
- [x] get the value has_products_ok an set into a var on template
- [x] refactory of product_ids field in product_public_category
- [x]  added a test to test has_products_ok field

### CODE REVIEW

**Comment by hugho-ad** [comment]
https://github.com/Vauxoo/yoytec/pull/396

File: website_product_filters/controllers/main.py [hunk 1]
```diff
@@ -94,9 +94,10 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
 
     def _child_has_products(self, category):
         if category.child_id:
+            print "No MAME!! \°/"
```
Comment (nhomar) [inline website_product_filters/controllers/main.py]: Ok no MAMO !!!!!! eejejejejejejej es más bonito estos simpáticos mensajes con ##

Comment (moylop260) [inline website_product_filters/controllers/main.py]: jajajaja

Comment (hugho-ad) [inline website_product_filters/controllers/main.py]: Jajajajajaja hay no mame! Jaja se me fue jajajajaja

**Comment by hugho-ad** [comment]
@moylop260 I've been trying do the next code, but the field has_products_ok does not fill up correctly

``` python
    @api.depends('product_ids')
    @api.multi
    def _get_has_products_ok(self):
        ids_with_products = []
        for record in self:
            if record.product_ids:
                ids_with_products.append(record.id)
        ids = self.browse(ids_with_products)
        ids.write({'has_products_ok': True})
```

**Comment by moylop260** [comment]
@hugho-ad 
I don't have a answer... but If you help me to add a unittest I can check the sql-log

**Comment by hugho-ad** [comment]
@moylop260 

Los cambios que hicimos, no esta surtiendo efecto
image

Haciendo que en el test tenga inconsistencias.

**Comment by moylop260** [comment]
@hugho-ad 
Do you have a news of this pr?

**Comment by hugho-ad** [comment]
@moylop260 
I've been working in other things, actually in this pr https://github.com/Vauxoo/yoytec/pull/313, when I finished, I'll work on this.
Regards

**Comment by nhomar** [comment]
@hugho-ad any news?

**Comment by hugho-ad** [comment]
@moylop260 @nhomar 
Added test, and refactory fields in product_public_category, using `store = True` and sensitive `api.depends('product_ids')`

I'm working on  fields type function with cycle by record with hard calculation in module website_product_avalability.

@nhomar , please read comments on this :https://github.com/Vauxoo/yoytec/issues/460#event-479300561, in order to consider the posibles changes on features in website_product_avalability

**Comment by hugho-ad** [comment]
@moylop260 this is ready to merge, improvements already done, 
Improvements for website_product_avalailability will be done in other pr

Regards

**Comment by moylop260** [comment]
Iijjj merge without dummy

**Pull request merged**